project/BuildInfo.scala

```diff
@@ -0,0 +1,288 @@
+                      |""".stripMargin,
+        target = DebBuildTarget(Seq("stretch"), "main", "bash,libcurl3,libre2-3,libunwind8", Seq.empty)),
+
+      MuslBuild.deb("ubuntu-older", Seq("trusty", "utopic", "vivid", "wily"), "main", "bash"),
+
+      BuildInfo(
+        name = "ubuntu-16-04",
+        baseImage = "ubuntu:16.04",
+        install = s"""|RUN \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install apt-transport-https && \\
+                      |  echo "deb https://dl.bintray.com/sbt/debian /" > /etc/apt/sources.list.d/sbt.list && \\
+                      |  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 2EE0EA64E40A89B84B2DF73499E82A75642AC823 && \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install bc build-essential openjdk-8-jre-headless ca-certificates-java clang++-3.8 libcurl4-openssl-dev libgc-dev libre2-dev libunwind8-dev sbt
+                      |""".stripMargin,
+        target = DebBuildTarget(Seq("xenial"), "main", "bash,libre2-1v5,libunwind8,libcurl3", Seq.empty)),
+
+      BuildInfo(
+        name = "ubuntu-16-10",
+        baseImage = "ubuntu:16.10",
+        install = s"""|RUN \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install apt-transport-https && \\
+                      |  echo "deb https://dl.bintray.com/sbt/debian /" > /etc/apt/sources.list.d/sbt.list && \\
+                      |  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 2EE0EA64E40A89B84B2DF73499E82A75642AC823 && \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install bc build-essential openjdk-8-jre-headless ca-certificates-java clang-3.8 libcurl4-openssl-dev libgc-dev libre2-dev libunwind8-dev sbt
+                      |""".stripMargin,
+        target = DebBuildTarget(Seq("yakkety"), "main", "bash,libre2-2,libunwind8,libcurl3", Seq.empty)),
+
+      BuildInfo(
+        name = "ubuntu-17-04_17-10",
+        baseImage = "ubuntu:17.04",
+        install = s"""|RUN \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install apt-transport-https && \\
+                      |  echo "deb https://dl.bintray.com/sbt/debian /" > /etc/apt/sources.list.d/sbt.list && \\
+                      |  apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 2EE0EA64E40A89B84B2DF73499E82A75642AC823 && \\
+                      |  apt-get -y update && \\
+                      |  apt-get -y install bc build-essential openjdk-8-jre-headless ca-certificates-java clang-3.8 libcurl4-openssl-dev libgc-dev libre2-dev libunwind8-dev sbt
+                      |""".stripMargin,
+        target = DebBuildTarget(Seq("zesty", "artful"), "main", "bash,libre2-3,libunwind8,libcurl3", Seq.empty)))
```

Comment:
To support a new distro, add it to this list. Note that some distros can reuse previous `BuildInfo` objects, i.e. note that `17.04` and `17.10` share an entry. I suspect as `re2` (which is the most problematic lib) matures, the number of `BuildInfo` objects we need will decrease.